src/mono/mono/mini/interp/jiterpreter.c

```diff
@@ -751,11 +753,12 @@ trace_info_get (gint32 index) {
 static gint32
 trace_info_alloc () {
 #ifdef DISABLE_THREADS
-	gint32 index = trace_count++,
+	gint32 index = trace_count++;
 #else
-	gint32 index = atomic_fetch_add ((atomic_int *)&trace_count, 1),
+	static_assert (sizeof(atomic_int) == sizeof(gint32) && ATOMIC_INT_LOCK_FREE == 2);
+	gint32 index = atomic_fetch_add ((atomic_int *)&trace_count, 1);
```

Comment:
Would it be more desirable to use the target instead of the target's type for the assert?

For example:

```c
static_assert (sizeof(atomic_int) == sizeof(trace_count) && ATOMIC_INT_LOCK_FREE == 2);
```